main/src/library/DelayMap.flix

```diff
@@ -166,7 +166,8 @@ mod DelayMap {
                 if (useParallelEvaluation(m))
                     let h = (k, v) -> g(k, force v);
                     let DMap(t) = m;
-                    RedBlackTree.parCount(threads() - 1, h, t)
+                    let threads = unchecked_cast(Environment.handle(threads)() as _ \ {});
```

Comment:
magnus has the final decision, but as matt suggested, if we put these two lines into a helper function, and use unchecked_cast around its body I think it could be okay. That function would then doc comment about why its okay to cast it away (System + IO is unobservable because performance is not observable in our model)

Reply:
hmm its used with multiple methods, not as easy

Reply:
I agree with @JonathanStarup 

Add a PRIVATE function to each of DelayList, DelayMap, and RedBlackTree:

```
def threads(): Int32 = unsafe ...
```

and use it there, so we don't have to use the Env effect.

Reply:
(i mean technically its not sound at that scope because the integer returned very much is dependent on Environment, its only after calling the relevant function and threads goes out of scope that the only observable effect is performance)